src/main/java/hudson/plugins/clover/CloverPublisher.java

```diff
@@ -136,9 +149,23 @@ static File getCloverXmlReport(Run<?, ?> build) {
         return new File(build.getRootDir(), "clover.xml");
     }
 
+    /**
+     * Gets the directory where the Clover Report is stored for the given build with a specific reportId.
+     */
+    static File getCloverXmlReport(Run<?, ?> build, String reportId) {
+        String fileName = (reportId == null || reportId.trim().isEmpty()) 
```

Comment:
There are several warnings about lack of test coverage for changes. Please double check if your unit tests handle them.